### PR TITLE
Full setup db

### PR DESCRIPTION
**Pull request by williamh890**

### CODE REVIEW

File: cloudformation/tropo/templates/hyp3_random_pass.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
+
```
Comment (Askaholic) [inline cloudformation/tropo/templates/hyp3_random_pass.py:5]: Where are the docstrings!!??

File: cloudformation/tropo/templates/hyp3_rds.py [hunk 2]
```diff
@@ -44,7 +44,6 @@ def get_security_group(name: str):
 # https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/Concepts.DBInstanceClass.html
 hyp3_db = t.add_resource(rds.DBInstance(
     "Hyp3DB",
-    DBInstanceIdentifier="hyp3-in-a-box",
```
Comment (Askaholic) [inline cloudformation/tropo/templates/hyp3_rds.py]: Why do you hate names so much? If I am going to spin up someone else's cloudformation template in my AWS Account, I want to be able to clearly see where resources are coming from. I don't want to have 5 RDS's all named `uihnkbvgcyitsudfg`. If name conflicts are an issue, then just make it a parameter with default value.

File: lambdas/random_password/src/random_password.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+            'Reason': 'Successfully generated a random string'
+        }
+
+    def _get_random_string(self):
+        length = self._get_length()
+
+        characters = string.digits + string.ascii_letters
+
+        random_string = ''.join(
+            random.SystemRandom().choice(characters) for _ in range(length)
```
Comment (Askaholic) [inline lambdas/random_password/src/random_password.py:29]: WOw s0 SEcUr3!

File: lambdas/setup_db/src/init_db.py [hunk 4]
```diff
@@ -8,30 +8,76 @@
...
+
+def add_db_admin_user(db):
+    user, password, db_name = [
+        os.environ[f'Hyp3DB{k}'] for k in ('User', 'Pass', 'Name')
+    ]
+
+    add_user_sql = text(f"""
+        CREATE USER hyp3_user WITH PASSWORD :password;
+        GRANT ALL PRIVILEGES ON DATABASE {db_name} to {user};
+    """.format(db, user))
```
Comment (Askaholic) [inline lambdas/setup_db/src/init_db.py]: You never want to do string formatting on something which will later be run as SQL. This is how SQL injections happen! Always use prepared statements, and parameters.

File: lambdas/setup_db/src/init_db.py [hunk 5]
```diff
@@ -8,30 +8,76 @@
...
+        add_user_sql,
+        password=password
     )
+
+
+def make_tables(db):
     Base.metadata.create_all(db.engine)
+
+
+def make_hyp3_admin_user(db):
```
Comment (Askaholic) [inline lambdas/setup_db/src/init_db.py:96]: Presumably this will be parameterized later. Maybe add a TODO comment as a reminder.

**Review by Askaholic — CHANGES_REQUESTED** [review]

File: lambdas/setup_db/src/init_db.py (hunk 4 above)
Comment (williamh890) [reply lambdas/setup_db/src/init_db.py]: It throw an error when using the prepared statement version. This was a hack to get the merge to pass the tests.

File: cloudformation/tropo/templates/hyp3_rds.py (hunk 2 above)
Comment (williamh890) [reply cloudformation/tropo/templates/hyp3_rds.py]: I just took that out cuz I was lazy and it was erroring the stack, I was planning on making an environment variable to use parameterized names or not in the template.

File: lambdas/random_password/src/random_password.py (hunk 3 above)
Comment (williamh890) [reply lambdas/random_password/src/random_password.py:29]: lel

**Pull request merged**